src/simulation/dynamics/prescribedLinearTranslation/prescribedLinearTranslation.cpp

```diff
@@ -0,0 +1,344 @@
+    PrescribedTranslationMsgPayload prescribedTranslationMsgOut;
+
+    prescribedTranslationMsgOut = PrescribedTranslationMsgPayload();
+
+    linearTranslationRigidBodyIn = LinearTranslationRigidBodyMsgPayload();
+    if (this->linearTranslationRigidBodyInMsg.isWritten()) {
+        linearTranslationRigidBodyIn = this->linearTranslationRigidBodyInMsg();
+    }
+
+    // This loop is entered (a) initially and (b) when each translation is complete.
+    // The parameters used to profile the translation are updated in this statement.
+    if (this->linearTranslationRigidBodyInMsg.timeWritten() <= callTime && this->convergence) {
+        // Update the initial time as the current simulation time
+        this->tInit = callTime * NANO2SEC;
+
+        // Store the reference scalar position
+        this->transPosRef = linearTranslationRigidBodyIn.rho;
+
+        this->transPosInit = this->transPos;
+
+        // Set the convergence to false until the translation is complete
+        this->convergence = false;
+
+        // Set the parameters required to profile the translation
+        if (this->coastOptionRampDuration > 0.0) {
+            this->computeCoastParameters();
+        } else {
+            this->computeParametersNoCoast();
+        }
+    }
+
+    double t = callTime * NANO2SEC;
+
+    // Compute the scalar translational states at the current simulation time
+    if (this->coastOptionRampDuration > 0.0) {
+        if (this->isFirstRampSegment(t)) {
+            this->computeFirstRampSegment(t);
+        } else if (this->isInCoastSegment(t)) {
+            this->computeCoastSegment(t);
+        } else if (this->isInSecondRampSegment(t)) {
+            this->computeSecondRampSegment(t);
+        } else {
+            this->computeTranslationComplete();
+        }
+    } else {
+        if (this->isInFirstRampSegmentNoCoast(t)) {
+            this->computeFirstRampSegment(t);
+        } else if (this->isInSecondRampSegmentNoCoast(t)) {
+            this->computeSecondRampSegment(t);
+        } else {
+            this->computeTranslationComplete();
+        }
+    }
+
+    // [m] Translational body position relative to the Mount frame expressed in M frame components
+    Eigen::Vector3d r_FM_M = this->transPos*this->transHat_M;
+    // [m/s] B frame time derivative of r_FM_M expressed in M frame components
+    Eigen::Vector3d rPrime_FM_M = this->transVel*this->transHat_M;
+    // [m/s^2] B frame time derivative of rPrime_FM_M expressed in M frame components
+    Eigen::Vector3d rPrimePrime_FM_M = this->transAccel*this->transHat_M;
+
+    eigenVector3d2CArray(r_FM_M, prescribedTranslationMsgOut.r_FM_M);
+    eigenVector3d2CArray(rPrime_FM_M, prescribedTranslationMsgOut.rPrime_FM_M);
+    eigenVector3d2CArray(rPrimePrime_FM_M, prescribedTranslationMsgOut.rPrimePrime_FM_M);
+    this->prescribedTranslationOutMsg.write(&prescribedTranslationMsgOut, this->moduleID, callTime);
```

Comment:
Add the line to write the payload to the C-wrapped output message.  See comment in *.h file.

Reply:
@patkenneally

Reply:
Done